pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -434,6 +419,32 @@ public AtomicLong getPendingWriteOps() {
         return pendingWriteOps;
     }
 
+    private CompletableFuture<Void> newTopicCompactionService() {
+        CompactionServiceFactory compactionServiceFactory = brokerService.pulsar().getCompactionServiceFactory();
+        return compactionServiceFactory.newTopicCompactionService(topic).thenAccept(topicCompactionService -> {
+            PersistentTopic.this.topicCompactionService = topicCompactionService;
+        });
+    }
```

Comment:
Is it better to move this method to PulsarService?

Reply:
done